### PR TITLE
feat(Portable): Combined Project Structure & First Example

### PR DESCRIPTION
**Pull request by kevinvangelder**
### Testing
- Universal project tested with universal tests
- Shared files tested by .Net46 tests via shared tests shared project, soon to also be tested by universal tests when NMock supports UWP (@matthargett PR)
- .Net46 files tested with .Net46 tests

### First Example
- Tracing implemented with null object pattern in .Net46
- NativeModuleBase (which depends on Tracing) moved to shared

cc: @GantMan

### CODE REVIEW

File: ReactWindows/ReactNative.Tests/project.json [hunk 1]
```diff
@@ -1,7 +1,8 @@
 ﻿{
   "dependencies": {
     "Facebook.CSSLayout": "2.0.1-pre",
-    "Microsoft.NETCore.UniversalWindowsPlatform": "5.2.2"
+    "Microsoft.NETCore.UniversalWindowsPlatform": "5.2.2",
+    "NUnit": "3.4.1"
   },
```
Comment (rozele) [inline ReactWindows/ReactNative.Tests/project.json]: Is this dependency being used yet?  I don't see any dependencies on ReactNative.Shared.Tests in ReactNative.Tests.csproj

File: ReactWindows/ReactNative.Net46/Tracing/Tracer.cs [hunk 2]
```diff
@@ -0,0 +1,38 @@
+﻿using System;
+
+namespace ReactNative.Tracing
+{
+    class NullDisposable : IDisposable
```
Comment (rozele) [inline ReactWindows/ReactNative.Net46/Tracing/Tracer.cs]: >class [](start = 4, length = 5)

struct

File: ReactWindows/ReactNative.Net46/Tracing/Tracer.cs [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        }
+    }
+    public class NullLoggingActivityBuilder
+    {
+        public IDisposable Start()
+        {
+            return new NullDisposable();
+        }
+    }
+    public static class Tracer
```
Comment (rozele) [inline ReactWindows/ReactNative.Net46/Tracing/Tracer.cs]: clean up whitespace.

File: ReactWindows/ReactNative/ReactNative.csproj [hunk 4]
```diff
@@ -113,7 +113,6 @@
     <Compile Include="Bridge\ReactInstance.cs" />
     <Compile Include="Bridge\DispatcherHelpers.cs" />
     <Compile Include="Bridge\JavaScriptBundleLoader.cs" />
-    <Compile Include="Bridge\NativeModuleBase.cs" />
     <Compile Include="Bridge\NativeModuleRegistry.cs" />
```
Comment (GantMan) [inline ReactWindows/ReactNative/ReactNative.csproj]: Agreed, it's a big bite.  The folder change kind of needed a small example to justify it (and really test the feasibility of) it.  It helped us see our path going forward, but yes it bloats this PR.  

`NativeModuleBase` seems to be one of the major fundamental lynchpins.  We're able to to finally move modules with this working.

**Comment by matthargett** [comment]
LGTM

**Comment by rozele** [comment]
>   </PropertyGroup>

This is the documentation info I'm talking about in ReactNative.Net46/Tracing/Tracer.cs

---
Refers to: ReactWindows/ReactNative/ReactNative.csproj:55 in dcb04cd. [](commit_id = dcb04cd6715b321cb1524269475837c87b4abf44, deletion_comment = False)

File: ReactWindows/ReactNative.Net46/Tracing/Tracer.cs [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+{
+    class NullDisposable : IDisposable
+    {
+        public void Dispose()
+        {
+
+        }
+    }
+
+    public class NullLoggingActivityBuilder
```
Comment (rozele) [inline ReactWindows/ReactNative.Net46/Tracing/Tracer.cs]: >NullLoggingActivityBuilder [](start = 17, length = 26)

Can you ensure you check the documentation box in the project settings for ReactNative.Net46 for all configurations and platforms?  We want to get compile time warnings for missing documentation on public surfaces like this.

Comment (rozele) [inline ReactWindows/ReactNative.Net46/Tracing/Tracer.cs]: As such, you'll have to either add docs to the public surfaces you've added here, or just remove `public` from Tracer and NullLoggingActivityBuilder 

---
In reply to: [82050122](https://github.com/ReactWindows/react-native-windows/pull/740#discussion_r82050122) [](ancestors = 82050122)

File: ReactWindows/ReactNative/ReactNative.csproj (hunk 4 above)
Comment (rozele) [inline ReactWindows/ReactNative/ReactNative.csproj]: Sounds good, thanks!

---
In reply to: [82005884](https://github.com/ReactWindows/react-native-windows/pull/740#discussion_r82005884) [](ancestors = 82005884)

File: ReactWindows/ReactNative.Net46/Tracing/Tracer.cs (hunk 5 above)
Comment (kevinvangelder) [inline ReactWindows/ReactNative.Net46/Tracing/Tracer.cs]: I can add the documentation. Removing `public` causes `insufficient privileges` errors on compile.

File: ReactWindows/ReactNative.Net46/Tracing/NullLoggingActivityBuilder.cs [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+        /// <returns>An empty disposable object.</returns>
+        public IDisposable Start()
+        {
+            return new NullDisposable();
+        }
+
+        /// <summary>
+        /// Empty disposable object.
+        /// </summary>
+        class NullDisposable : IDisposable
```
Comment (rozele) [inline ReactWindows/ReactNative.Net46/Tracing/NullLoggingActivityBuilder.cs]: You technically didn't need docs here :-)

File: ReactWindows/ReactNative.Net46/ReactNative.Net46.csproj [hunk 7]
```diff
@@ -4,18 +4,19 @@
...
+    <AssemblyName>ReactNative.Net46</AssemblyName>
     <TargetFrameworkVersion>v4.6</TargetFrameworkVersion>
     <FileAlignment>512</FileAlignment>
   </PropertyGroup>
   <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Debug|x86'">
     <PlatformTarget>x86</PlatformTarget>
     <OutputPath>bin\x86\Debug\</OutputPath>
     <DefineConstants>DEBUG</DefineConstants>
+    <DocumentationFile>bin\x86\Debug\ReactNative.Net46.XML</DocumentationFile>
   </PropertyGroup>
```
Comment (rozele) [inline ReactWindows/ReactNative.Net46/ReactNative.Net46.csproj]: Looks like you only enabled documentation for Debug|x86, when you check "XML Documentation File", make sure you've selected "All Configurations" and "All Platforms" from the top dropdown boxes in the settings.

File: ReactWindows/ReactNative.Net46/Tracing/NullLoggingActivityBuilder.cs (hunk 6 above)
Comment (kevinvangelder) [inline ReactWindows/ReactNative.Net46/Tracing/NullLoggingActivityBuilder.cs]: Right, because it's not public... got it. Want me to remove?

Comment (rozele) [inline ReactWindows/ReactNative.Net46/Tracing/NullLoggingActivityBuilder.cs]: Doesn't matter, up to you.  Less to maintain in the long term if you remove it.

---
In reply to: [82057270](https://github.com/ReactWindows/react-native-windows/pull/740#discussion_r82057270) [](ancestors = 82057270)

File: ReactWindows/ReactNative.Net46/Tracing/NullLoggingActivityBuilder.cs [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+        {
+            /// <summary>
+            /// Dispose method to conform to IDisposable.
```
Comment (rozele) [inline ReactWindows/ReactNative.Net46/Tracing/NullLoggingActivityBuilder.cs]: >IDisposable [](start = 45, length = 11)

You may remove this anyway, but you can make this a link using <see cref="IDisposable" />

**Comment by rozele** [comment]
:shipit:

**Pull request merged**